taxonomy/utils.py

```diff
@@ -81,29 +83,39 @@ def get_product_skill_model_and_identifier(product_type):
     """
     Return the Skill Model along with its identifier based on product type.
     """
-    return (ProgramSkill, 'program_uuid') if product_type == ProductTypes.Program else (CourseSkills, 'course_key')
+    if product_type == ProductTypes.Program:
+        return (ProgramSkill, 'program_uuid')
+    if product_type == ProductTypes.XBlock:
+        return (XBlockSkills, 'usage_key')
+    if product_type == ProductTypes.XBlockThrough:
+        return (XBlockSkillThrough, 'xblock_id')
+    return (CourseSkills, 'course_key')
 
 
-def update_skills_data(key_or_uuid, skill_external_id, confidence, skill_data, product_type):
+def update_skills_data(key_or_uuid, skill_external_id, confidence, skill_data, product_type, **kwargs):
     """
-    Persist the skills data in the database either for Program or Course.
+    Persist the skills data in the database for Program, Course or XBlock.
     """
-    skill, __ = Skill.objects.update_or_create(external_id=skill_external_id, defaults=skill_data)
+    skill, _ = Skill.objects.update_or_create(external_id=skill_external_id, defaults=skill_data)
+    if product_type == ProductTypes.XBlock:
+        x_model, x_identifier = get_product_skill_model_and_identifier(product_type)
+        product_type = ProductTypes.XBlockThrough
+        xblock, _ = x_model.objects.update_or_create(
+            **{x_identifier: key_or_uuid},
+            defaults={"hash_content": kwargs.get("hash_content"), "auto_processed": True},
+        )
+        key_or_uuid = xblock.id
+    if is_skill_blacklisted(key_or_uuid, skill.id, product_type):
+        return
     skill_model, identifier = get_product_skill_model_and_identifier(product_type)
-    kwargs = {
-        identifier: key_or_uuid,
-        'skill': skill,
-    }
-    defaults = {
-        'confidence': confidence
-    }
-    if not is_skill_blacklisted(key_or_uuid, skill.id, product_type):
-        _, created = skill_model.objects.update_or_create(**kwargs, defaults=defaults)
-        action = 'created' if created else 'updated'
-        LOGGER.error(f'{skill_model} {action} for key {key_or_uuid}')
+    kwargs = {identifier: key_or_uuid, "skill": skill}
+    defaults = {"confidence": confidence}
+    _, created = skill_model.objects.update_or_create(**kwargs, defaults=defaults)
```

Comment:
We can either go with additionally updating keyword arguments(kwargs) or default and then just call update_or_create just once.